### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Kateryna-Bordonos17**

### CODE REVIEW

**Review by RVChornyy — APPROVED** [review]
Nice!

**Review by wQuelS — CHANGES_REQUESTED** [review]
изображение
Your formatting style in the first function differs from style in two other functions. Please, try to be consistent with it and provide a fix

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+        "path": file_path,
+        "status": "failed" if len(errors) > 0 else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key, errors)
```
Comment (Andrii-Silenko) [inline app/main.py]: Maybe "key" is not the best name of varuable. I'd prefer "file_path" or "path_of_file

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(errors[i])
+                   for i in range(len(errors))],
+        "path": file_path,
+        "status": "failed" if len(errors) > 0 else "passed"
```
Comment (Andrii-Silenko) [inline app/main.py:16]: It's ok, but you can rewrite this condition as:
```suggestion
        "status": "failed" if len(errors) > 0 else "passed"
```

Comment (Andrii-Silenko) [inline app/main.py]: You can rewrite this condition as:
```suggestion
        "status": "failed" if errors else "passed"
```

**Review by Andrii-Silenko — APPROVED** [review]

**Review by RVChornyy — APPROVED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(errors[i])
```
Comment (fsocie7y) [inline app/main.py:13]: in cases like this u can use `for element in elements: ...` without indexes

**Review by fsocie7y — APPROVED** [review]

**Pull request closed without merging**